pkg/app/mutate_app.go

```diff
@@ -124,26 +122,34 @@ func (m *Mutator) MutateApp(ctx context.Context, oldApp, app v1alpha1.App, opera
 		}
 	}
 
+	var patchLabels bool
+
+	labelPatches, err := m.mutateLabels(ctx, app, appVersionLabel)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	if len(labelPatches) > 0 {
+		result = append(result, labelPatches...)
+		patchLabels = true
+	}
+
 	ver, err := semver.NewVersion(appVersionLabel)
 	if err != nil {
 		m.logger.Debugf(ctx, "skipping mutation of app %#q in namespace %#q due to version label %#q", app.Name, app.Namespace, appVersionLabel)
 		return nil, nil
 	}
 
 	// If the app CR does not have the unique version and is < 3.0.0 we skip
-	// the defaulting logic. This is so the admission controller is not enabled
-	// for existing platform releases.
+	// the defaulting logic apart from the labels. This is so the admission
+	// controller is not enabled for existing platform releases.
 	if key.VersionLabel(app) != uniqueAppCRVersion && ver.Major() < 3 {
-		m.logger.Debugf(ctx, "skipping mutation of app %#q in namespace %#q due to version label %#q", app.Name, app.Namespace, appVersionLabel)
-		return nil, nil
-	}
-
-	labelPatches, err := m.mutateLabels(ctx, app, appVersionLabel)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-	if len(labelPatches) > 0 {
-		result = append(result, labelPatches...)
+		if patchLabels {
+			m.logger.Debugf(ctx, "mutating only labels of app %#q in namespace %#q due to version label %#q", app.Name, app.Namespace, appVersionLabel)
+			return result, nil
+		} else {
```

Comment:
I think you don't need this `else` statement.

Reply:
I think I was still in holiday mode :) I'll fix.